fcrepo-http-commons/src/main/java/org/fcrepo/http/commons/exceptionhandlers/ExceptionDebugLogging.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright 2015 DuraSpace, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.fcrepo.http.commons.exceptionhandlers;
+
+import org.slf4j.Logger;
+
+/**
+ * @author barmintor
+ * @since 6/28/16
+ */
+public interface ExceptionDebugLogging {
+    /**
+     * Log a Throwable at the DEBUG level, log the stacktrace at the TRACE level.
+     *
+     * @param context ExceptionDebugLogging the exception intercepting context.
+     * @param error Throwable the intercepted error.
+     * @param logger Logger the logger to use
+     */
+    public default void debugException(ExceptionDebugLogging context, Throwable error, Logger logger) {
+        if (!logger.isDebugEnabled()) {
```

Comment:
Would you like me to comment the 2 accessor vs 1 and early-exit
justification there?

On Wed, Jun 29, 2016 at 11:05 AM, A. Soroka <notifications@github.com>
wrote:

> In
> fcrepo-http-commons/src/main/java/org/fcrepo/http/commons/exceptionhandlers/ExceptionDebugLogging.java
> <https://github.com/fcrepo4/fcrepo4/pull/1060#discussion_r68963196>:
>
> > +import org.slf4j.Logger;
> > +
> > +/**
> > + * @author barmintor
> > + * @since 6/28/16
> > + */
> > +public interface ExceptionDebugLogging {
> > +    /**
> > +     * Log a Throwable at the DEBUG level, log the stacktrace at the TRACE level.
> > +     *
> > +     * @param context ExceptionDebugLogging the exception intercepting context.
> > +     * @param error Throwable the intercepted error.
> > +     * @param logger Logger the logger to use
> > +     */
> > +    public default void debugException(ExceptionDebugLogging context, Throwable error, Logger logger) {
> > +        if (!logger.isDebugEnabled()) {
>
> It may very well be. It's all an exception path anyway. I really wanted to
> mention it as a general point, because I am a pretentious pettifogging
> pedant.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/fcrepo4/fcrepo4/pull/1060/files/fea09f811935c7410ffef5f6a9541025b8229d54#r68963196>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAHUNIq5xYBwiR8-ckROGuc3oQLlaxcMks5qQomtgaJpZM4JBMtM>
> .
>